src/System.Reflection.Metadata/tests/Metadata/Decoding/DisassemblingTypeProvider.cs

```diff
@@ -144,22 +156,22 @@ public virtual string GetByReferenceType(string elementType)
             return elementType + "&";
         }
 
-        public virtual string GetGenericMethodParameter(int index)
+        public virtual string GetGenericMethodParameter(DisassemblingGenericContext genericContext, int index)
         {
-            return "!!" + index;
+            return "!!" + (genericContext == null ? index.ToString() : genericContext.MethodParameters[index]);
```

Comment:
>genericContext [](start = 27, length = 14)

Is this ever used with null context? It would be nice to have context always non-null -- that kind of demonstrates that we are flowing it everywhere it might be needed and is always available when needed.